test/c++/nda_mpi.cpp

```diff
@@ -110,6 +113,29 @@ TEST(Arrays, MPI) { //NOLINT
 
 // --------------------------------------
 
+TEST(Arrays, MPIReduceCustom) { //NOLINT
+
+  mpi::communicator world;
+  using matrix_t = nda::matrix<double>;
+  using arr_t = nda::array<matrix_t, 1>;
+
+  arr_t a(7);
+  arr_t r_exact(7);
+  for (int i = 0; i < a.extent(0); ++i){
+    a(i) = matrix_t{4, 4};
+    a(i)(k_, l_) << world.rank() * (k_ + l_);
+
+    r_exact(i) = matrix_t{4, 4};
+    r_exact(i)(k_, l_) << (world.size() - 1) * world.size() / 2 * (k_ + l_);
+  }
+
+  arr_t r = mpi::all_reduce(a, world);
+
+  //EXPECT_ARRAY_EQ(r, r_exact); // FIXME Does not compile
```

Comment:
The TEST does not test anything !

Reply:
I actually adjusted the test in a later commit. I will merge these changes together.

Reply:
ok, do I need to see this commit ?

Reply:
https://github.com/TRIQS/nda/pull/10/commits/02b5f1c809779190151e1b8e057abb5c27dbdb5b